src/Build/BackEnd/Components/SdkResolution/SdkResolverManifest.cs

```diff
@@ -21,11 +21,44 @@ internal class SdkResolverManifest
         internal static SdkResolverManifest Load(string filePath)
         {
             using (FileStream fs = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read))
-            using (XmlDictionaryReader reader = XmlDictionaryReader.CreateTextReader(fs, new XmlDictionaryReaderQuotas()))
             {
-                DataContractSerializer serializer = new DataContractSerializer(typeof(SdkResolverManifest));
-                return (SdkResolverManifest)serializer.ReadObject(reader, true);
+                try
+                {
+                    var doc = new XmlDocument();
+                    doc.Load(fs);
```

Comment:
I'm curious if going one step further down to XmlReader helps at all. This file is so trivial. 

(I wonder if we even need it to be xml, we just need a single string to be defined and it could have just been a text file with that string.)

Reply:
XmlReader makes detection for duplicates a bit less trivial, where most of the cost was for spinning up reflection. I compared both, timing was similar, with less readable code.

Reply:
That surprises me also but I can't argue with numbers.